OpenLiberty/open-liberty
# Issue 10519: Add automatic certificate checker for ACME

Fixes #10519 

- Added a scheduler to check for expiration or revocation
- Added locking around renew/revoke
- Added a timer window around REST renewal

Tests added:
- Scheduler renews expiring certificate
- Scheduler renews revoked certificate
- Scheduler doesn't run if disabled
- Back to back renewal
- Scheduler stops when acmeCA feature is removed, restarts on addition
- Scheduler runs after bad config is updated to valid config
- Stop Pebble during test, ensure we reschedule the certchecker and renew cert that's expiring

Test to add in another PR: Opened #12328 to address
- Enable/disable the cert checker during runtime (set `certCheckerSchedule` to 0), make sure it stops/starts appropriately
- Call renew while certchecker is going -- disable auto? What are we checking for here?
- Call revoke and renew at the same time?
- Stop Pebble during test, ensure we reschedule the certchecker and renew cert that's revoked

For #9017
